src/v/kafka/server/handlers/handler_probe.h

```diff
@@ -59,6 +68,8 @@ class handler_probe {
 
     uint64_t _bytes_received{0};
     uint64_t _bytes_sent{0};
+
+    std::optional<log_hist_public> _latency{std::nullopt};
```

Comment:
why is it optional?

Reply:
`_probes` is a sparsely populated array so I was using `optional` with the `hdr_hist` to avoid a large allocation for probes that didn't point to any Kafka handler. 

`log_hist` still makes an allocation for it's vector of buckets that is avoided with this optional too. However, it's much smaller of an allocation so maybe it's worth the wasted space to remove the optional?

Reply:
I feel like if the sparseness is important it should be handled one level up with an `optional<probe>` or `unique_ptr<probe>` as elements of the array rather having each member need to worry about whether it's small when unused. Is this also why `initialize()` call is there? It also means we need to have this invariant that all probes are initialized before they are used which requires more thought.

I think it's fine for this change though.

Reply:
Handling the spareness one level up is definitely something I'll do in a follow up PR. Beyond that I'm also considering switching the `vector` that holds the buckets in `log_hist` to an array and letting the user decide whether to store the histogram inline with whatever probe class they have or using a `std::unique_ptr<log_hist>` to add a level of indirection and to store it in a separate allocation. Now that the histogram is at most 208 bytes storing it inline with other class members shouldn't be too problematic. Thoughts?